app/src/main/java/se/rebeccazadig/bokholken/login/UserViewModel.kt

```diff
@@ -25,6 +24,21 @@ class UserViewModel : ViewModel() {
         loginRepo.logOutInRepo()
     }
 
+    fun saveUser() {
+        inProgress.value = true
+
+        viewModelScope.launch {
+            val userId = userId.value.orEmpty()
+            val userInfo = userName.value.orEmpty()
+            val userContact = userContact.value.orEmpty()
+            val userCity = userCity.value.orEmpty()
+            val user = User(name = userInfo, contact = userContact, city = userCity, id = userId)
```

Comment:
Consider a situation when ALL fields are empty. Do you want to save such a user?

Reply:
val userId = userId.value.orEmpty() - doesn't that mean that userId will always be ""?

Reply:
That's weird. I don't have the same code in my app. In my app I have val userid = loginRepo.getUserId(). The others does have an orEmpty and I'll keep that on userName (because I didn't make that mandatory to fill in) but I'll remove that on the others. On the other hand.. I don't think I have to since I disabled the button if the fields are left blank. But yeah its unnecessary code for sure.